### PR TITLE
add e2e startup test for rootless k3s

### PR DESCRIPTION
**Pull request by osodracnai**
#### Proposed Changes ####
Tests for rootless k3s

#### Verification ####
`go test -timeout=35m ./tests/e2e/rootless/...`

#### Linked Issues ####
 https://github.com/k3s-io/k3s/issues/8051

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: tests/e2e/rootless/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,79 @@
+ENV['VAGRANT_NO_PARALLEL'] = 'no'
+NODE_ROLES = (ENV['E2E_NODE_ROLES'] ||
+  ["server-0"])
+NODE_BOXES = (ENV['E2E_NODE_BOXES'] ||
+  ['generic/ubuntu2204', 'generic/ubuntu2204'])
```
Comment (dereknola) [inline tests/e2e/rootless/Vagrantfile:5]:
```suggestion
  ['generic/ubuntu2204'])
```

File: tests/e2e/rootless/rootless_startup_test.go [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+var ci = flag.Bool("ci", false, "running on CI")
+var local = flag.Bool("local", false, "deploy a locally built K3s binary")
+
+// Environment Variables Info:
+// E2E_RELEASE_VERSION=v1.27.1+k3s2 or nil for latest commit from master
+
+// This test suite is used to verify that K3s can start up with dynamic configurations that require
+// both server and agent nodes. It is unique in passing dynamic arguments to vagrant, unlike the
+// rest of the E2E tests, which use static Vagrantfiles and cluster configurations.
+// If you have a server only flag, the startup integration test is a better place to test it.
```
Comment (dereknola) [inline tests/e2e/rootless/rootless_startup_test.go:26]:
```suggestion
// Rootless is only valid on a single node, but requires node/kernel configuration, requiring a E2E test environment.
```

**Review by dereknola — CHANGES_REQUESTED** [review]
Obviously WIP, but these were some initial comments

File: tests/e2e/rootless/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  vm.hostname = role
+  # An expanded netmask is required to allow VM<-->VM communication, virtualbox defaults to /32
+  vm.network "private_network", ip: "#{NETWORK_PREFIX}.#{100+node_num}", netmask: "255.255.255.0"
+
+  vagrant_defaults = '../vagrantdefaults.rb'
+  scripts_location = Dir.exists?("./scripts") ? "./scripts" : "../scripts"
+  load vagrant_defaults if File.exists?(vagrant_defaults)
+  
+  defaultOSConfigure(vm)
+  dockerInstall(vm)
```
Comment (brandond) [inline tests/e2e/rootless/Vagrantfile:26]: why are we installing docker for rootless?

File: tests/e2e/rootless/Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  addCoverageDir(vm, role, GOCOVER)
+  install_type = getInstallType(vm, RELEASE_VERSION, GITHUB_BRANCH)
+
+  node_ip = "#{NETWORK_PREFIX}.#{100+node_num}"
+
+
+  vm.provision "shell", inline: "ping -c 2 k3s.io"
+
+  if role.include?("server")
+    dockerInstall(vm)
```
Comment (brandond) [inline tests/e2e/rootless/Vagrantfile:36]: this appears to be installing docker a second time, on servers?

File: tests/e2e/rootless/Vagrantfile (hunk 3 above)
Comment (osodracnai) [reply tests/e2e/rootless/Vagrantfile:26]: testing startup with docker

Comment (brandond) [reply tests/e2e/rootless/Vagrantfile:26]: so we're testing rootless, and rootless with docker?

Comment (osodracnai) [reply tests/e2e/rootless/Vagrantfile:26]: yes

Comment (brandond) [reply tests/e2e/rootless/Vagrantfile:26]: I don't know that we've tested that previously, or if this is something we actually want to support.

Normally when running rootless, containerd also runs within the unprivileged user namespace. If using rootless + docker, the system docker daemon will still run as root, which kind of misses the point of rootless.

**Review by dereknola — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Comment by brandond** [comment]
Needs to be rebased to remove s390x from CI

**Review by dereknola — APPROVED** [review]

**Pull request merged**